Oracle.md

```diff
@@ -0,0 +1,196 @@
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the signed number to obtain the actual outcome value.
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+unit: Celcius degrees
+precision: 0
+```
+
+### Large range
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using numerical decomposition.
+
+The event descriptor should include:
+
+- base: a number representing the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed (e.g. 2 for binary, 10 for decimal, 16 for hex-decimal).
+- is-signed: a boolean value indicating whether the outcomes can be negative.
+- r-values: an array of R values, one for each of the digit that will be signed by the oracle, with the first value of the array being used for signing the leftmost digit. If the is-signed value is true, an additional R-value must be provided as the first element in this array, which will be used to sign the string "+" in case of an outcome with a positive value or zero value and the string "-" in case of an outcome with a negative one.
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base exponent by which to multiply the number represented by the composition of the digits to obtain the actual outcome value.
+
+In the case where the number of digits that make up the outcome value exceeds the number of r-values that the oracle committed to, the oracle should sign the maximum possible value that he can attest to. 
+
+When the number of digits that make up the outcome is lower than the number of r-values that the oracle committed to, the oracle should sign and publish a signature over the number zero with the extra R-values.
+
+
+#### Example: BTC/USD rate
+
+The following example illustrates how numerical decomposition can help reduce the number of required CETs while covering a large range of possible outcomes.
+
+Alice and Bob wish to enter into a DLC at time `t` when BTC/USD ~= $10000 with a maturity time of `t+1`.
+The oracle provides 6 R-values, meaning that it can attest to outcomes in the range `[0,999999]`:
+
+```
+base: 10
+isSigned: false
+RValues: [R0, R1, R2, R3, R4, R5]
+unit: dollars
+precision: 0
+```
+
+Alice and Bob create a DLC with "relevant" outcomes between $10000 and $10999.
+Below $10000 Alice gets everything, and above $10999 Bob gets everything.
+They use the `R0` and `R1` values to cover the cases where BTC/USD < $10000 using `R0=0` and `R1=0`.
+They use `R0`, `R1` and `R2` to cover the cases where BTC/USD > $10999 and <= $19999, using `R1=0`, `R1=1` and `R2 in [0-9]`.
+They use `R0` and `R1` to cover the cases where BTC/USD >= $20000 and < $99999, using `R0=0` and `R1 in [2,9]`.
+Finally, they use only `R0` to cover the cases where BTC/USD >= $100000 and <= $999999, using `R0 in [1-9]`.
```

Comment:
You use `Ri` interchangeably for a nonces and digit values, I suggest you switch digit values to `Di`.

Reply:
or Mi

Reply:
Good point, updating the example to use `D0-5` and explain a bit better (at least I hope)